Content.Shared/Doors/Systems/SharedDoorSystem.Firelock.cs

```diff
@@ -32,79 +34,66 @@ public override void Initialize()
-        var overrideAccess = (args.User != null) && _accessReaderSystem.IsAllowed(args.User.Value, uid);
+        var overrideAccess = (args.User != null) && _accessReaderSystem.IsAllowed(args.User.Value, firelock);
 
-        if (!component.Powered || (!overrideAccess && component.IsLocked))
+        if (!firelock.Comp.Powered || !overrideAccess && firelock.Comp.IsLocked)
             args.Cancel();
         else if (args.User != null)
-            WarnPlayer((uid, component), args.User.Value);
-    }
-
-    private void OnBeforePry(EntityUid uid, FirelockComponent component, ref BeforePryEvent args)
-    {
-        if (args.Cancelled || !component.Powered || args.StrongPry || args.PryPowered)
-            return;
-
-        args.Cancelled = true;
+            WarnPlayer(firelock, args.User.Value);
     }
 
-    private void OnDoorGetPryTimeModifier(EntityUid uid, FirelockComponent component, ref GetPryTimeModifierEvent args)
+    private void OnDoorGetPryTimeModifier(Entity<FirelockComponent> firelock, ref GetPryTimeModifierEvent args)
     {
-        WarnPlayer((uid, component), args.User);
+        WarnPlayer(firelock, args.User);
 
-        if (component.IsLocked)
-            args.PryTimeModifier *= component.LockedPryTimeModifier;
+        if (firelock.Comp.IsLocked)
+            args.PryTimeModifier *= firelock.Comp.LockedPryTimeModifier;
     }
 
-    private void WarnPlayer(Entity<FirelockComponent> ent, EntityUid user)
+    private void WarnPlayer(Entity<FirelockComponent> firelock, EntityUid user)
     {
-        if (ent.Comp.Temperature)
+        if (firelock.Comp.Temperature)
         {
             _popupSystem.PopupClient(Loc.GetString("firelock-component-is-holding-fire-message"),
-                ent.Owner,
+                firelock.Owner,
                 user,
                 PopupType.MediumCaution);
         }
-        else if (ent.Comp.Pressure)
+        else if (firelock.Comp.Pressure)
         {
             _popupSystem.PopupClient(Loc.GetString("firelock-component-is-holding-pressure-message"),
-                ent.Owner,
+                firelock.Owner,
                 user,
                 PopupType.MediumCaution);
         }
     }
 
-    private void OnAfterPried(EntityUid uid, FirelockComponent component, ref PriedEvent args)
+    private void OnAfterPried(Entity<FirelockComponent> firelock, ref PriedEvent args)
     {
-        component.EmergencyCloseCooldown = _gameTiming.CurTime + component.EmergencyCloseCooldownDuration;
+        firelock.Comp.EmergencyCloseCooldown = _gameTiming.CurTime + firelock.Comp.EmergencyCloseCooldownDuration;
     }
 
     #endregion
 
     #region Visuals
 
-    private void UpdateVisuals(EntityUid uid, FirelockComponent component, EntityEventArgs args) => UpdateVisuals(uid, component);
+    private void UpdateVisuals(EntityUid uid, FirelockComponent component, EntityEventArgs args) =>
+        UpdateVisuals(uid, component);
 
     private void UpdateVisuals(EntityUid uid,
```

Comment:
This function's signature could be changed to

```csharp
    private void UpdateVisuals(Entity<FirelockComponent> firelock)
```
, as it's only ever called from event subscriptions which ensure there's a firelock component _and_ they never resolve the other components ahead of time.

IF you do this, you can change the `resolve` guard to
```csharp
        if (!TryComp<DoorComponent>(firelock, out var door) ||
            !TryComp<AppearanceComponent>(firelock, out var appearance))
            return;
```

Reply:
Coming back to this, since this is actually only used from event subscriptions, it could just follow the usual event subscription handler pattern of taking an `events` arg that's discarded.